assets/js/common/anonymous-user.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+define(['require',
+        'angular',
+        'angularfire',
+        'common/firebase-service'], function(require) {
+
+    angular
+      .module('usersFactoryModule', [
+      'firebase',
+      'voteFireRefServiceModule'
+      ])
+      .factory("AnonymousUser",
+        function(
+          $firebase,
+          voteFireRef
+        ) {
+          return function() {
+            var ref = voteFireRef()
+            var authResult = {}
+            if (! ref.getAuth()) {
+              ref.authAnonymously(function(error, authData) {
+                if (error) {
+                  authResult["error"] = true;
+                }
+                authResult["voterId"] = authData.uid
+              });
+            } else{
+              authResult["voterId"] = ref.getAuth().uid
+            }
+
+            return authResult;
```

Comment:
What about returning a promise?
I have a concern that code inside `ref.authAnonymously` will fill the `authResult` in the future. But the code using it, doesn't check if data is available or not yet. So `votenow` method is going to fail if user clicked too early!